mne/decoding/tests/test_transformer.py

```diff
@@ -229,3 +229,49 @@ def test_unsupervised_spatial_filter():
     usf = UnsupervisedSpatialFilter(PCA(4), average=True)
     usf.fit_transform(X)
     assert_raises(ValueError, UnsupervisedSpatialFilter, PCA(4), 2)
+
+
+def test_temporal_filterer():
+    """Test methods of TemporalFilter."""
+    X = np.random.rand(10, 5, 2)
+
+    # Add tests for different combinations of l_freq and h_freq
+    filt = TemporalFilter(l_freq=5, h_freq=15, sfreq=100,
+                          filter_length='auto', l_trans_bandwidth=None,
+                          h_trans_bandwidth=None)
+    with warnings.catch_warnings(record=True):  # stop freq attenuation warning
+        result = filt.fit_transform(X)
+        assert_true(X.shape == result.shape)
+        assert_array_equal(filt.fit(X).transform(X), result)
+
+    filt = TemporalFilter(l_freq=None, h_freq=15, sfreq=100,
+                          filter_length='auto', l_trans_bandwidth=None,
+                          h_trans_bandwidth=None)
+    with warnings.catch_warnings(record=True):  # stop freq attenuation warning
+        X = filt.fit_transform(X)
+
+    filt = TemporalFilter(l_freq=1, h_freq=1, sfreq=100)
+    with warnings.catch_warnings(record=True):  # stop freq attenuation warning
+        assert_raises(ValueError, filt.fit_transform, X)
+
+    filt = TemporalFilter(l_freq=5, h_freq=None, sfreq=100,
+                          filter_length='auto', l_trans_bandwidth=None,
+                          h_trans_bandwidth=None)
+    with warnings.catch_warnings(record=True):  # stop freq attenuation warning
+        result = filt.fit_transform(X)
```

Comment:
just loop all the tests above to use much fewer lines: e.g.
```
for low, high in ((None, 15.), (5., None) , ...):
     filt = TemporalFilter(low, high, sfreq=100,
                                     filter_length='auto', l_trans_bandwidth=None,
                                     h_trans_bandwidth=None)
     filt.fit_transform(X)
```

same for the raises.